x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -5839,6 +5845,241 @@ public void testLookupStats() {
         );
     }
 
+    //
+    // Lookup JOIN
+    //
```

Comment:
I wanted to double check the pushdown happens inside the plan and since our CSV tests don't support (yet) lookup queries in non-Lucene mode, I've added added a bunch of testing her (such as the case with rename or breaking the filter up).